prdoc/schema_user.json

```diff
@@ -0,0 +1,224 @@
+            "nullable": false,
+            "title": "Database Migration",
+            "description": "List of the Database Migrations or empty array: []",
+            "items": {
+              "$ref": "#/$defs/migration_db"
+            },
+            "minItems": 0,
+            "required": [
+              "name",
+              "description"
+            ]
+          },
+          "runtime": {
+            "type": "array",
+            "title": "Runtime Migration",
+            "nullable": false,
+            "description": "List of the Runtime Migrations or empty array: []",
+            "minItems": 0,
+            "items": {
+              "$ref": "#/$defs/migration_runtime"
+            },
+            "required": [
+              "db",
+              "runtime"
+            ]
+          }
+        }
+      },
+      "host_functions": {
+        "title": "Host Functions",
+        "description": "List of the host functions involved in this PR.",
+        "type": "array",
+        "items": {
+          "$ref": "#/$defs/host_function"
+        }
+      }
+    },
+    "required": [
+      "title",
+      "doc",
+      "crates"
+    ],
+    "$defs": {
+      "audience": {
+        "description": "You may pick one or more audiences and address those users with appropriate documentation, information and warning related to the PR.",
+        "oneOf": [
+          {"const": "Node Dev",
+          "title": "Node Dev",
+          "description": "Those who build around the client side code. Alternative client builders, SMOLDOT, those who consume RPCs. These are people who are oblivious to the runtime changes. They only care about the meta-protocol, not the protocol itself."},
+
+          {"const": "Runtime Dev",
+          "title": "Runtime Dev",
+          "description": "All of those who rely on the runtime. A parachain team that is using a pallet. A DApp that is using a pallet. These are people who care about the protocol (WASM), not the meta-protocol (client)."},
+
+          {"const": "Node Operator",
+          "title": "Node Operator",
+          "description": "Those who don't write any code and only run code."},
+
+          {"const": "Runtime User",
+          "title": "Runtime User",
+          "description": "Token holders who don't run anything but are involved in the network."}
+        ]
+      },
+      "crate": {
+        "type": "object",
+        "description": "You have the option here to provide a semver level as hint to help with the publishing of crates.",
+        "additionalProperties": false,
+        "properties": {
+          "name": {
+            "type": "string"
+          },
+          "semver": {
+            "$ref": "#/$defs/semver"
+          },
+          "note": {
+            "type": "string"
+          }
+        }
+      },
+      "semver": {
+        "type": "string",
+        "enum": [
+          "major",
+          "minor",
+          "patch",
+          "no change"
+        ]
+      },
```

Comment:
I thought about this again. We probably want to have the versions numbers in the master branch any way. So, why not just bump the version in the crates directly? 

We would require one CI job that ensures that we don't bump a version multiple times, before we do a release again. However, this should be fairly simple.

CC @Morganamilo

Reply:
I'm not a fan of this idea as it's less clearly visible and lacks an explicit "no change" option.

Reply:
Why is it less clearly visible? I mean the version is directly bumped in the `Cargo.toml`.

Reply:
It's hard to see if a crate was bumped, what kind of bump, if it's been bumped for multiple reasons, what PR the bump was associated with etc. And it lacks an explicit "this crate was changed but not in a way that changes anything".

Thinking from a tooling perspective it will be easier to scan PRdocs. Both for the automated releases and for displaying what changed via some sort of status command.

Reply:
> It's hard to see if a crate was bumped, what kind of bump

This is an easy git diff between the latest release and master. Not really complicated.

> , if it's been bumped for multiple reasons,

Between releases there should be only one bump of the major, or one of minor and one of patch. If there was a bump of patch and then minor, patch is reverted etc.

In generally people are already wanting to know what is the latest crate version in master. This would not be solved by adding this stuff to prdoc.



> Both for the automated releases and for displaying what changed via some sort of status command.

This is a git blame with some more logic.

Reply:
I'm fine if we may mention the "affected" crates, if that makes things easier (while I still think that this could be automated).